### PR TITLE
RF: save() in line with #546

### PR DESCRIPTION
**Pull request by hanke**
Also combination with 'store_true' didn't make sense.

### CODE REVIEW

**Comment by hanke** [comment]
Missing tests for `GitRepo.tag()` and `save(files=...)`

File: datalad/interface/save.py [hunk 1]
```diff
@@ -81,11 +91,39 @@ def __call__(message, dataset=None, auto_add_changes=True, version_tag=None):
...
-            ds.repo.add('.', git=True)
-        ds.repo.commit(message)
+            files = [ds.path]
+
+        if files:  # could still be none without auto add changes
+            for f in files:
+                absf = abspath(f)
+                # XXX Is there a better way to handle files in mixed repos?
+                ds.repo.add(absf)
+                ds.repo.add(absf, git=True)
```
Comment (yarikoptic) [inline datalad/interface/save.py]: Couldn't you pass a list of files? One at a time will be slow!

File: datalad/support/gitrepo.py [hunk 2]
```diff
@@ -1125,6 +1125,17 @@ def update_submodule(self, path, mode='checkout', init=False):
         cmd += ['--', path]
         self._git_custom_command('', cmd)
 
+    def tag(self, tag):
```
Comment (yarikoptic) [inline datalad/support/gitrepo.py]: Add a message kwarg... ideally we should use/facilitate annotated tags

File: docs/examples/3rdparty_analysis_workflow.sh [hunk 3]
```diff
@@ -77,7 +77,7 @@ bash code/run_analysis.sh ||true
 
 datalad unlock
 bash code/run_analysis.sh
-datalad save -a "Alice always helps"
+datalad save -a -m "Alice always helps"
```
Comment (yarikoptic) [inline docs/examples/3rdparty_analysis_workflow.sh]: Should the be tagging example?

Comment (hanke) [inline docs/examples/3rdparty_analysis_workflow.sh]: Will come later when actual narrative arrives. I was planing to do this prior last publish.

File: datalad/support/gitrepo.py (hunk 2 above)
Comment (hanke) [inline datalad/support/gitrepo.py]: Will do later. This one is just to have a single git custom command.

File: datalad/interface/save.py (hunk 1 above)
Comment (hanke) [inline datalad/interface/save.py]: Looks like those functions support that -- will do.

**Comment by hanke** [comment]
@bpoldrack @yarikoptic I'd appreciate a soonish merge. I promise tests for GitRepo.tag() later -- when it actually does something reasonable.

**Pull request merged**